### PR TITLE
[CLD-6219] Add Company Name field to Purchase Modal

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
Add a Company Name field to the purchase modal that's required, so it can be passed to CWS for export compliance screening. This field will be auto populated with any previously entered value, and will update the Stripe customer with the new value. 

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-6219

#### Screenshots
image

#### Release Note

```release-note
None
```

### CODE REVIEW

**Review by JulienTant — APPROVED** [review]
LGTM @nickmisasi

File: webapp/channels/src/components/payment_form/payment_form.tsx [hunk 1]
```diff
@@ -161,6 +164,20 @@ export default class PaymentForm extends React.PureComponent<Props, State> {
...
             paymentDetails = (
                 <React.Fragment>
+                    <div className='form-row'>
+                        <Input
+                            name='company_name'
+                            type='text'
+                            value={this.state.company_name}
+                            onChange={this.handleInputChange}
+                            onBlur={this.onBlur}
+                            placeholder={Utils.localizeMessage(
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/payment_form/payment_form.tsx:174]: Can you please try to use formatMessage with useIntl hoc for this component, we are trying to reduce dependency on these custom functions

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by yasserfaraazkhan** [comment]
company name is appearing I stripe and in the community channel

File: webapp/channels/src/components/payment_form/payment_form.tsx (hunk 1 above)
Comment (nickmisasi) [reply webapp/channels/src/components/payment_form/payment_form.tsx]: I opted to just convert the component to a functional one, been meaning to for a while anyways. Switched to formatMessage from react-intl.

**Comment by nickmisasi** [comment]
@JulienTant @yasserfaraazkhan would you guys mind having another look? I've changed up the payment_form after some feedback from Zubair.

**Review by M-ZubairAhmed — APPROVED** [review]
Thank you for changing Nick

**Review by yasserfaraazkhan — APPROVED** [review]
working as expected [(web hook link)](https://community.mattermost.com/private-core/pl/7rd4ti9tk38pmbixq8i3k4je3a)

**Review by gabrieljackson — APPROVED** [review]

**Review by JulienTant — APPROVED** [review]
Second review, all good :)

**Pull request merged**